src/GitHub.InlineReviews/PullRequestStatusBarPackage.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Threading;
+using System.Runtime.InteropServices;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.ComponentModelHost;
+using GitHub.Helpers;
+using GitHub.Services;
+using GitHub.VisualStudio;
+using GitHub.InlineReviews.Services;
+using Task = System.Threading.Tasks.Task;
+
+namespace GitHub.InlineReviews
+{
+    [Guid(Guids.PullRequestStatusPackageId)]
+    [PackageRegistration(UseManagedResourcesOnly = true, AllowsBackgroundLoading = true)]
+    [ProvideAutoLoad(Guids.GitSccProviderId, PackageAutoLoadFlags.BackgroundLoad)]
+    public class PullRequestStatusBarPackage : AsyncPackage
+    {
+        /// <summary>
+        /// Initialize the PR status UI on Visual Studio's status bar.
+        /// </summary>
+        protected override async Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
+        {
+            var usageTracker = (IUsageTracker)await GetServiceAsync(typeof(IUsageTracker));
+            var serviceProvider = (IGitHubServiceProvider)await GetServiceAsync(typeof(IGitHubServiceProvider));
+            var componentModel = (IComponentModel)await GetServiceAsync(typeof(SComponentModel));
+            var pullRequestSessionManager = componentModel.DefaultExportProvider.GetExport<IPullRequestSessionManager>();
```

Comment:
MEF calls need to happen on the main thread.

Also, it would be better if everything goes through `GitHubServiceProvider`. This way we can mock all these types (and even bypass MEF entirely if we want to) without the caller having to change any calls. `GitHubServiceProvider` already does look ups via ComponentModel, and the MEF lookups ensure that the results are loaded from our DLLS and not any other type that might be overriding our stuff, so that's a plus.

Ideally, though, we don't use MEF in package initialization, because loading the catalog for the first time is really expensive. Since you need to grab `IVSGitExt` here, I suggest you expose it as a service in the `ServiceProviderPackage` class like we do the other services like metrics, that way you'll get an instance directly with no need for MEF scanning everything.

Reply:
Of course, if `IPullRequestSessionManager` has too many dependencies to be exposed as a service, we can go through MEF, but should always do it via the `GitHubServiceProvider` anyway.

Reply:
I've fixed it so that everything goes through `GitHubServiceProvider`. No more explicit dependency on `IComponentModel`.